### DISCUSSION TITLE
MCS for SMARTS?

### DISCUSSION THREAD

**Discussion by simonmb** [rdkit/rdkit#4665] [Q&A] [answered]
Hi, I know it is far more complex than with SMILES, but I was wondering what would be the easiest way to find overlapping substructures for SMARTS instead of SMILES.

```python
patt1 = Chem.MolFromSmarts('[CH0](=O)O')
patt2 = Chem.MolFromSmarts('CC(=O)C')
patt3 = Chem.MolFromSmarts('C[CH](=O)C')

... Here is where the magic happens

MCS_SMARTS_patt1_patt2 = 'C=O'
MCS_SMARTS_patt1_patt3 = 'C'
```

**Comment by simonmb** [comment]
If one could enumerate all possible structures defined by a SMARTS pattern one could check for this.
Is there a function available in rdkit for doing this?

**Comment by greglandrum** [comment, marked as answer]
Doing SMARTS-SMARTS matches is really hard and the RDKit doesn't currently have functionality for that.

**Reply by simonmb** [reply]
Thanks for your answer, I already asked a similar question some time ago. I don't know if you are aware of this publication: https://pubs.acs.org/doi/10.1021/acs.jcim.9b00250

**Reply by greglandrum** [reply]
Yeah, that's a really nice paper and I would love to have that functionality in the RDKit, but it's a significant amount of work to implement that approach.
